oscarlab/graphene
# Protected argv

## Description of the changes 

_This is an RFC PR. After we converge on the final version I'll implement a similar PR for envs._

This PR implements a way to pass runtime arguments from a file, with intention to use it in conjunction with either trusted or protected files.

Unfortunately, argv passing is quite ugly because of `loader.exename` and very limited manifest syntax. After fixing #870 and #1053 the implementation and usage could be simplified.

One known issue with the current code is that argv of children goes through the untrusted host under Linux-SGX PAL (through `execve` arguments, see #1561).

This PR is a breaking change and there's (AFAIK) no better way to implement it - the old manifest syntax was insecure-by-default (allowing unstrusted args). Users will need to use `loader.insecure__use_cmdline_argv` if they don't want to use the file arguments.
For apps which are intended to run without any special arguments, the recommended way is to set `loader.execname` and don't pass any additional arguments. This will get a bit more intuitive after allowing inline argv in the manifest (see a comment in the code).

This resolves a half of #508.

## How to test this PR? 

New LibOS test included. Also, verify if this doesn't break any examples you use but aren't tested in Jenkins (or aren't committed to the repo).


---
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/oscarlab/graphene/1562)
